charms/modeldb-backend/reactive/modeldb_backend.py

```diff
@@ -55,7 +56,36 @@ def start_charm():
                         {
                             'name': 'config',
                             'mountPath': '/config-backend/',
-                            'files': {'config.yaml': hookenv.config('config')},
+                            'files': {
+                                'config.yaml': yaml.dump(
+                                    {
+                                        'grpcServer': {'host': hookenv.service_name(), 'port': grpc_port},
+                                        'database': {
+                                            'DBType': 'rdbms',
+                                            'RdbConfiguration': {
+                                                'RdbDatabaseName': 'mariadb',
+                                                'RdbDriver': 'com.mysql.cj.jdbc.Driver',
+                                                'RdbDialect': 'org.hibernate.dialect.MySQL5Dialect',
+                                                'RdbUrl': f'jdbc:mysql://{mysql.host()}:{mysql.port()}',
+                                                'RdbUsername': 'root',
+                                                'RdbPassword': hookenv.config('mysql-root-password'),
+                                            },
+                                        },
+                                        'artifactStore_grpcServer': {
+                                            'host': store.all_joined_units[0].application_name,
+                                            'port': store.all_joined_units[0].received['port'],
```

Comment:
These should be better encapsulated by the interface layer and exposed as proper attributes / methods.  Also, using the application name as seen from the receiving side will break with cross-model relations; instead, the sending side should include the host info as part of the relation data.  That would also allow this to support the this coming from a non-K8s backer, though I don't know if that's really relevant in this case.